### ISSUE TITLE
Printed document has shifts in colors

### ISSUE THREAD

**Issue by detrist** [pibooth/pibooth#227]
Hi All,

Not sure whether it is a bug or not but I have a really weird situation what so far I have not resolved. I have a HiTi P525L printer which seems working fine from Windows 10 by printing images. 

For some reason on RPI4 the printed images from Pibooth have shifts in colors along the X axis. The same happens if I want to print the image from shell so this is not the problem of the Pibooth code for first look. However, the the result is fine if I ask GThumb on RPI4 to print the same picture without changing its default settings. Definitly GThumb does something more which makes it work. Please see the below example for a better understanding of my issue:

![IMG_20210418_001704](https://user-images.githubusercontent.com/605377/115128087-a338e080-9fdb-11eb-8d34-24e837d9168d.jpg)

Have you ever had similar issue about the printing? Any suggestions would be appriciated.

Thanks for your feedback in advance.

**Comment by anxuae** [comment]
Hi @detrist

It seems an issue concerning the support of your printer in CUPS (print server not provided by pibooth).
Did you try to install/update [GutenPrint](http://gimp-print.sourceforge.net/index.php).

``
sudo apt-get install printer-driver-gutenprint
``

Did you tray to change the settings of your printer in the [CUPS admin page](http://localhost:631/printers/)?

BR
anxuae

**Comment by detrist** [comment]
Hi @anxuae,

I will try it again with a fresh PI installation but for first look it did not help and so I decided to move the printing part into Windows temporarly by providing a Rest API and call it instead of CUPS. Meanwhile we are in contact with the printer support provider if they could provide us a better driver.

I will update this issue later I have any.

Many thanks, detrist

**Comment by detrist** [comment]
Better late than never, I finally managed to make the printing work with the gutenprint driver, but there were some tricks in its installation steps:
1.) Probably optional but I deleted the "printer-driver-gutenprint" package I found in the RPI repo and then downloaded the latest snapshot version from SourceForge[https://sourceforge.net/projects/gimp-print/files/snapshots/](url)
2.) I had to delete the official CUPS drivers and filters from the disk I previously installed

Also I noticed that the official driver contained a PPD file which was made with CUPS 1.4 and during the build of the driver a lot of deprication notice appeared. Still waiting for update from the product support.

Thank you for your assistance.
